lib/graphql/static_validation/literal_validator.rb

```diff
@@ -10,28 +10,56 @@ def initialize(context:)
 
       def validate(ast_value, type)
         if ast_value.is_a?(GraphQL::Language::Nodes::NullValue)
-          !type.kind.non_null?
+          maybe_raise_if_invalid(ast_value) do
+            !type.kind.non_null?
+          end
         elsif type.kind.non_null?
-          (!ast_value.nil?) && validate(ast_value, type.of_type)
+          maybe_raise_if_invalid(ast_value) do
+            (!ast_value.nil?)
+          end && validate(ast_value, type.of_type)
         elsif type.kind.list?
           item_type = type.of_type
           ensure_array(ast_value).all? { |val| validate(val, item_type) }
         elsif ast_value.is_a?(GraphQL::Language::Nodes::VariableIdentifier)
           true
         elsif type.kind.scalar? && constant_scalar?(ast_value)
-          type.valid_input?(ast_value, @context)
-        elsif type.kind.enum? && ast_value.is_a?(GraphQL::Language::Nodes::Enum)
-          type.valid_input?(ast_value.name, @context)
+          maybe_raise_if_invalid(ast_value) do
+            type.valid_input?(ast_value, @context)
+          end
+        elsif type.kind.enum?
+          maybe_raise_if_invalid(ast_value) do
+            if ast_value.is_a?(GraphQL::Language::Nodes::Enum)
+              type.valid_input?(ast_value.name, @context)
+            else
+              # if our ast_value isn't an Enum it's going to be invalid so return false
+              false
+            end
+          end
         elsif type.kind.input_object? && ast_value.is_a?(GraphQL::Language::Nodes::InputObject)
-          required_input_fields_are_present(type, ast_value) &&
-            present_input_field_values_are_valid(type, ast_value)
+          maybe_raise_if_invalid(ast_value) do
+            required_input_fields_are_present(type, ast_value)
+          end && present_input_field_values_are_valid(type, ast_value)
         else
-          false
+          maybe_raise_if_invalid(ast_value) do
+            false
+          end
         end
       end
 
       private
 
+
+      def maybe_raise_if_invalid(ast_value)
+        ret = yield if block_given?
```

Comment:
What's the use case for this method when there's no `block_given?` If it's never used that way, can we remove this check to make its intention a bit clearer?

Reply:
i'll remove it - i guess it's just habit to add that.

Reply:
Sure, I guess another option would be to _handle_ the no-block case, for example, by raising a descriptive error. That way, a future user would understand the intention there. I just wasn't sure if it was _supposed_ to work `if !block_given?` in this case, be cause it would _return_!

Reply:
i agree that it shouldn't ever be called without a block so i just removed the ✔️